src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -61,7 +61,19 @@ Outputs a fingerprint of its contents and its size in bytes, separated by a spac
 protos for each directory found. Outputs a fingerprint of the canonical top-level Directory proto \
 and the size of the serialized proto in bytes, separated by a space.",
               )
-              .arg(Arg::with_name("source").required(true).takes_value(true)),
+              .arg(
+                Arg::with_name("source")
+                  .required(true)
+                  .takes_value(true)
+                  .help(
+                    "Path of directory to actually ingest. Must either be relative (to `root`) or \
+absolute and a child of `root`.",
+                  ),
+              )
+              .arg(Arg::with_name("root").long("root").takes_value(true).help(
+                "Root under which source lives. The Directory proto produced will be relative to \
+this directory.",
```

Comment:
Clarified (it doesn't actually default to cwd).

The idea is that if your filesystem looks like:
/foo
/foo/bar
/foo/baz

then specifying source=/foo will give you a directory: {bar, baz}.

I haven't found a good way of wording this in the help-text... Any suggestions are very welcome.

Reply:
Made `root` a required arg. If even just the two of us have different mental models of what the expected behaviour here is, we should probably just make people be explicit to avoid the possibility for ambiguity :)